server/src/main/java/org/elasticsearch/index/query/QueryRewriteContext.java

```diff
@@ -465,4 +466,22 @@ public void setQueryRewriteInterceptor(QueryRewriteInterceptor queryRewriteInter
         this.queryRewriteInterceptor = queryRewriteInterceptor;
     }
 
+    public boolean isMetadataField(String field) {
+        return mapperService.isMetadataField(field);
+    }
+
+    public boolean isMultiField(String field) {
+        if (runtimeMappings.containsKey(field)) {
+            return false;
+        }
+        return mapperService.isMultiField(field);
+    }
+
+    public NestedLookup nestedLookup() {
+        return mappingLookup.nestedLookup();
+    }
+
+    public boolean hasMappings() {
+        return mappingLookup.hasMappings();
+    }
```

Comment:
these methods rely on state that's already available in query rewrite context, there's no good reason to expose them only in SearchExecutionContext, that's more of a leftover from before we were able to rewrite based on mappings only.